File: anplayer-plugin/src/main/java/com/xeinebiu/anplayer/plugin/media/extractor/extractor/AlbumExtractor.kt
package com.xeinebiu.anplayer.plugin.media.extractor.extractor

import com.xeinebiu.anplayer.plugin.media.extractor.exception.AlbumNotFoundException
import com.xeinebiu.anplayer.plugin.media.extractor.model.Album
import com.xeinebiu.anplayer.plugin.media.extractor.model.descriptor.MediaDescriptor

/**
 * Extractor for a single Album
 */
abstract class AlbumExtractor : Extractor<MediaDescriptor>() {

    /**
     * Returns the album
     */
    @Throws(AlbumNotFoundException::class)
    abstract fun getAlbum(): Album
}

File: src/lib/extractor/album.extractor.ts
import { Extractor } from './extractor';
import { Album, AlbumDetail, Media } from '../model';

/**
 * Extractor for the contents of a single album, season, or collection.
 *
 * Returned by `AnPlayerExtractor.getAlbumExtractor(album)`. The root
 * `next()` method yields the items inside the collection (tracks, episodes,
 * films, or nested sub-albums).
 */
export type AlbumExtractor = Extractor<
    Media | Album,
    {
        /**
         * Returns the full detail record for this album (title, description,
         * artwork, ratings, etc.).
         *
         * Throw `AlbumNotFoundError` if the album no longer exists.
         */
        getAlbum(): Promise<AlbumDetail>;
    }
>;
